reactor-test/src/main/java/reactor/test/publisher/ColdTestPublisher.java

```diff
@@ -226,6 +221,76 @@ void onComplete() {
 			parent.remove(this);
 			actual.onComplete();
 		}
+
+		private void drain(long n) {
```

Comment:
drain has to be guarded by a WIP in order to avoid parallel drains (since there is no restriction on which thread calls `next` from the `TestPublisher` API vs `request` from the `Subscription` API)